gapic-generator/lib/gapic/model/mixins.rb

```diff
@@ -55,9 +55,7 @@ def mixins
 
       # @return [Enumerable<String>] Full proto names of the mix-in services
       def mixin_services
-        @mixin_services ||= services_in_config.select do |service|
-          service != LRO_SERVICE && !@api_services.include?(service)
-        end
+        @mixin_services ||= (services_in_config & SERVICE_TO_DEPENDENCY.keys)
```

Comment:
the guard in checking @api_services was that when we generate e.g. Location it might be mentioned in its own config and we wouldn't want it to generate as a mixin to itself. It should be kept I think.

Reply:
Good point. I didn't think about that case. Will add it back.

Reply:
So I don't think checking against `@api_services` will work because services may be configured to include protos related to their mixins. (For example, my test case, privateca/v1, has location protos in its build file even though location is meant to be a mixin.) So instead, I tested the gem name against the dependencies required if the service is treated as a mixin, which should handle this case.